src/api/groups.ts

```diff
@@ -0,0 +1,79 @@
+import { Response, Router } from 'express';
+import { GroupRequest as Request } from '../typings';
+import { Group } from '../models/group.model';
+import { IGroup, IGroupBaseAttributes } from '../types/group';
+
+// TODO: ADD REQUEST VALIDATION FOR PERMISSIONS
+
+export const router = Router();
+type IGroupLocals = Record<'group', IGroup>;
+
+router.param('id', async (req: Request, _, next, id) => {
+	console.log('>>>>>> entered group id param ', id);
+	req.group = (await Group.findByPk(id)) ?? undefined;
+
+	next();
+});
+
+router
+	.route('/:id')
+	.get(async (req: Request<{ id: string }, IGroup, {}, {}, IGroupLocals>, res: Response) => {
+		if (req.group) {
+			const group = req.group;
+			const users = await group.$get('users');
```

Comment:
Why don't you use proposed on previous step project architecture with Service layer, incapsulating business logic?

Reply:
resolved in PR for HW5